notes/shelley/CoreDIF.md

```diff
@@ -0,0 +1,153 @@
+  - We don't have topology change problems?
+  - These cause you to end up with loops which is bad for resource use.
+  - Don't have problems because we have state and so can deal with loops.
+- If we choose a new route, do we create a partition? Can we reason about
+  that so as to prove that we don't create partitions?
+- Neil will think about it.
+  Looking for the right piece of literature about this.
+
+- Make a best guess about a route: you know all of the pairwise delta-Q.
+  You pick (some deterministically, some randomly) who to send it to in the
+  hope that, according to the delta-Q's you know, it will optimally go
+  throughout the network.
+
+- Matrix of delta-Q's (100 x 100). How to fill it up? We know our own, we
+  propagate our own throughout the network (must propagate in order to
+  propagate, hm...)
+  I'm about to publish. I look at the matrix and I can compute paths which
+  with some certainty will get the data to everybody within some time. Of
+  course, it depends upon what the forwarders do.
+  Minimize maximum delay.
+  Also, due to paranoia, send to a few others with some random choice.
+
+- How long before we time-out a peer i.e. set its delta-Q to _|_ ?
+
+- Complication: some of the nodes are relays, some are stakeholders.
+  One network entity per public key????? We need to identify the nodes.
+  Would it not work for a core node to simply claim that its addresses is
+  that of a relay that it controls?
+
+### Unrelated notes
+
+- Benchmarks for deserializing: include in CI?
+- We want numbers to ensure that our logic layer time expenses fit within
+  the time budgets required for diffusion RE slot time.
+- It will be much easier to gather these measurements once we cut out networking
+  from logic.
+
+TIME is such an issue. Everything is limited by time and it causes so much
+trouble.
+Seeing the same issue when thinking about real-time music composition and
+performance. It's hard enough for a single computer to make time guarantees,
+let alone a network of them.
+
+## Questions after meeting with Neil and Duncan November 8
+
+1. Nodes in the core DIF never hold coins? Only stake? Why? Does this buy any
+   security at all? I suppose the machine itself is a target, and you wouldn't
+   want to keep the private key to the _coins_ on that machine. Keeping the
+   key to the delegate is fine, because the holder of the coins can revoke if
+   there's a compromise. Correct?
+
+2. Ability to deny being delegated to? Is this a must have? I would hope so.
+
+3. What does sharing "on the blockchain" really mean? How does it work? I'm
+   very skeptical. Why not flood the (non-core-dif) network with your bearer
+   address encrypted with the target's public key?
+
+4. Is the core dif construction mechanism essentially the same as the bigger
+   network? Start with some XOR kademlia-style thing, then use delta-Q to
+   refine it. Only difference is the wholistic approach, in which the entire
+   weighted graph is known?
```

Comment:
I'm not aware of anything contradicting that. And if we can make them as similar as possible that is obviously good, we can reuse logic & code.